superset-frontend/src/dashboard/util/getRelatedCharts.ts

```diff
@@ -103,7 +103,7 @@ export function getRelatedCharts(
         ),
       };
     }
-    if (!filter.filterType) {
+    if (!filter.filterType && Object.keys(slices).includes(filterKey)) {
```

Comment:
nit: we could probably simplify `Object.keys(slices).includes(filterKey)` to just `slices[filterKey]`